statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,721 @@
+
+        return hess_arr
+
+    def predict(self, params, exog=None, exog_infl=None, exposure=None,
+                offset=None, which='mean'):
+        """
+        Predict response variable of a count model given exogenous variables.
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+        exog : array, optional
+            A reference to the exogenous design.
+            If not assigned, will be used exog from fitting.
+        exog_infl : array, optional
+            A reference to the zero-inflated exogenous design.
+            If not assigned, will be used exog from fitting.
+        offset : array, optional
+            Offset is added to the linear prediction with coefficient equal to 1.
+        exposure : array, optional
+            Log(exposure) is added to the linear prediction with coefficient
+            equal to 1. If exposure is specified, then it will be logged by the method.
+            The user does not need to log it first.
+        which : string, optional
+            Define values that will be predicted.
+            'mean', 'poisson-mean', 'linear', 'mean-nonzero', 'prob-zero, 'prob'
+            Default is 'mean'.
+
+        Notes
+        -----
+        """
+        if exog is None:
+            exog = self.exog
+            offset = getattr(self, 'offset', 0)
+            exposure = getattr(self, 'exposure', 0)
+
+        if exog_infl is None:
+            exog_infl = self.exog_infl
+
+        if exposure is None:
+            exposure = 0
+        elif exposure != 0:
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = 0
+
+        params_infl = params[:self.k_inflate]
+        params_main = params[self.k_inflate:]
+
+        if self.infl == 'logit':
+            prob_poisson = 1 / (1 + np.exp(np.dot(exog_infl, params_infl)))
+        elif self.infl == 'probit':
+            raise NotImplemented('Predict for Probit inflation not implemented')
+
+        lin_pred = np.dot(exog, params_main[:self.exog.shape[1]]) + exposure + offset
+        
+        tmp_exog = self.model_main.exog
+        tmp_endog = self.model_main.endog
+        self.model_main.exog = exog
+        self.model_main.endog = np.zeros((exog.shape[0]))
+        llf = self.model_main.loglikeobs(params_main)
+        self.model_main.exog = tmp_exog
+        self.model_main.endog = tmp_endog
+
+        prob_zero = (1 - prob_poisson) + prob_poisson * np.exp(llf)
+
+        if which == 'mean':
+            return prob_poisson * np.exp(lin_pred)
+        elif which == 'poisson-mean':
+            return np.exp(lin_pred)
+        elif which == 'linear':
+            return lin_pred
+        elif which == 'mean-nonzero':
+            return prob_poisson * np.exp(lin_pred) / (1 - prob_zero)
+        elif which == 'prob-zero':
+            return prob_zero
+        elif which == 'prob':
+            return self._predict_prob(params)
```

Comment:
this should depend on arguments exog, exog_infl, ... provided to predict
or be a function of what already has been computed